### ISSUE TITLE
the messages that result from a TransactionRangeQuery should be passed directly to the channel for sending

### ISSUE THREAD

**Issue by woutslakhorst** [nuts-foundation/nuts-node#1309]
(Saves buffering in memory)

**Comment by woutslakhorst** [comment]
This is quite hard to do. A visitor pattern would be executed from within a Read transaction. The current conversion from TX to network DTO's isn't nested within a transaction. This would require passing db transactions from different layers..

**Comment by gerardsn** [comment]
I think this also helps with the outbox limit size of #1310. Currently all chunks are dumped in the outbox channel at once and if there are more chunks that the limit, the message cannot be sent in full.

If this happens more gradually by sending chunks as they are created, the nodes have some time to clear the outbox as it is filled. Slow nodes will still hit a limit here, but that is what the outbox limit was for anyway.

Not sure how the processing times of the different steps compare here though and if this will actually help.

**Comment by gerardsn** [comment]
@woutslakhorst is the 2 page limit per request enough to close this?

**Comment by woutslakhorst** [comment]
Yeah, we won't be investing in making this streaming with the current plans anyway
